stdlib/asm/crypto/stark/deep_queries.masm

```diff
@@ -467,11 +459,34 @@ end
 #!
 #! Input: [query_ptr, ...]
 #! Output: [...]
-#! Cycles: 6 + num_queries * 473
+#! Cycles: 24 + num_queries * 445
 export.compute_deep_composition_polynomial_queries
+
     exec.constants::fri_com_ptr
     dup.1
-    #=>[query_ptr, query_end_ptr, ...]
+    push.0
+    exec.constants::deep_rand_coef_ptr
+    exec.constants::ood_trace_ptr
+    exec.constants::current_trace_row_ptr
+    exec.constants::tmp9 mem_storew
+    #=>[X, query_ptr, query_end_ptr, ...]
+
+    # Compute the negations of z and gz.
+    # We do it here as this computation is common to all queries.
+    exec.constants::z_ptr mem_loadw
+    drop drop
+    neg swap neg  
+    #=> [-z0, -z1, ...]
+    dup.1 exec.constants::trace_domain_generator_ptr mem_load mul
+    #=> [-gz1, -z0, -z1, ...]
+    swap
+    #=> [-z0, -gz1, -z1, ...]
+    dup exec.constants::trace_domain_generator_ptr mem_load mul
+    #=> [-gz0, -z0, -gz1, -z1, ...]
+    swap
+    #=> [-z0, -gz0, -gz1, -z1, ...]
+    exec.constants::tmp1 mem_storew
```

Comment:
nit: I would add a comment above this line saying that we are saving `[-z1, -gz1, -gz0, -z0]` into memory location `tmp1`.

Reply:
Added